core-common/src/main/kotlin/com/hedvig/android/core/common/ApplicationScope.kt

```diff
@@ -0,0 +1,10 @@
+package com.hedvig.android.core.common
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.MainScope
+
+/**
+ * A [CoroutineScope] which lives for the entire lifecycle of the application. Can be used to launch top-level actions
+ * which need to out-live the current CoroutineScope, like even after a ViewModel would be killed.
+ */
+class ApplicationScope : CoroutineScope by MainScope()
```

Comment:
Serves as a top-level CoroutineScope to be used to run anything that is not tied to the specific ViewModel we're currently in or something like that, but things that should rather be run on a global level, even if we go back and kill the ViewModel for example. MainScope also has a SupervisorJob like viewModelScope does, which means that if any child coroutine fails that's okay and keeps this scope alive too.

This is what we had done in notification-service too https://github.com/HedvigInsurance/notification-service/blob/master/service/src/main/java/com/hedvig/notificationservice/utils/ApplicationScope.kt and it worked well for this use case.